### ISSUE TITLE
Fix memory pointer overwrite issue in BirdNet TFLite inference loop

### ISSUE THREAD

**Issue by coolsocket** [google-research/perch-hoplite#129]
In `perch_hoplite/zoo/models_tf.py`, when running `BirdNet` model inference in a loop over an `audio_array`, the `get_tensor()` method returns a reference to the internal TFLite buffer. Because this buffer is mutated in subsequent `model.invoke()` calls, all appended arrays in the `logits` and `embeddings` lists end up pointing to the exact same memory of the very last audio chunk. 

**Proposed Fix**: 
Add explicit copies to break the memory references:
```python
# Explicit copy to break references
logit_tensor = self.model.get_tensor(output_details['index'])
logit_copy = np.empty_like(logit_tensor)
logit_copy[:] = logit_tensor
logits.append(logit_copy)

emb_tensor = self.model.get_tensor(embedding_idx)
emb_copy = np.empty_like(emb_tensor)
emb_copy[:] = emb_tensor
embeddings.append(emb_copy)
```
This prevents downstream applications from receiving corrupted batch predictions.

**Comment by sdenton4** [comment]
Hello! Apologies for the slow response.
It looks like we can use a `.copy()` operation to reduce the line-count of the fix. Added this PR to do so; feel free to shout if it doesn't do the trick:

https://github.com/google-research/perch-hoplite/pull/132
